lib/Server.js

```diff
@@ -506,7 +507,7 @@ class Server {
         this.setupMagicHtmlFeature();
       },
       setup: () => {
-        if (typeof this.options.setup === 'function') {
+        if (this.options.setup) {
```

Comment:
No need check on `typeof` because we validate options and it is always will be Function https://github.com/webpack/webpack-dev-server/blob/master/lib/options.json#L313. Same for `before`/`after`/`onListening`

Reply:
Also, we can modify L528(`if (this.options.contentBase !== false) {`) ?